transcom/mymove
# B-18108 prime sim add shipment id when diversion box is checked

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A871301&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Mountain Movers are working on diverted shipments and in order to track the order of shipments that are being diverted, we need to know the parent shipment. @cameroncaci has been working on [this PR](https://github.com/transcom/mymove/pull/11673#pullrequestreview-1797252302) to allow for the `divertedFromShipmentId` parameter to be passed in, and is required, if `diversion` is `true`.

This PR is focused on updating the prime simulator to allow for that functionality to help with easier testing by the gov & our lovely testers.

The following changes were made:
- updated `PrimeUIShipmentCreateForm.jsx` with a new text input that appears when the diversion box is checked
- added in some validation to ensure that the uuid that is passed in is in a valid uuid format
- updated tests to reflect changes

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access MM in office app
2. Login as the prime or multi-role
3. Access any move and create a shipment
4. Check diversion box and you should see the text input box appear
5. Fill out the form and if you want, check out your debugger to see the value `divertedFromShipmentId` gets passed through to the `mto_shipments` handler

## Screenshots
Text input box appears when diversion box is checked
![Screenshot 2023-12-27 at 8 50 06 AM](https://github.com/transcom/mymove/assets/136510600/112abfc8-35c5-4123-8ba9-e94d0616385a)
